pandas/io/date_converters.py

```diff
@@ -1,35 +1,110 @@
+    """
+    Parse columns with years, months and days into a single date column.
+
+    .. deprecated:: 1.1.0
+        Use pd.to_datetime({"year": year_col, "month": month_col, "day": day_col})
+        instead to get a Pandas Series.
+        Use ser = pd.to_datetime({"year": year_col, "month": month_col, "day": day_col})
+        and np.array([s.to_pydatetime() for s in ser]) instead to get a Numpy array.
+    """
+    warnings.warn(
+        """
+        Use pd.to_datetime({"year": year_col, "month": month_col, "day": day_col}) instead to get a Pandas Series.
+        Use ser = pd.to_datetime({"year": year_col, "month": month_col, "day": day_col}) and
+        np.array([s.to_pydatetime() for s in ser]) instead to get a Numpy array.
+""",  # noqa: E501
+        FutureWarning,
+        stacklevel=2,
+    )
+
+    ser = to_datetime({"year": year_col, "month": month_col, "day": day_col})
+    return np.array([s.to_pydatetime() for s in ser])
 
 
 def parse_all_fields(year_col, month_col, day_col, hour_col, minute_col, second_col):
-    year_col = _maybe_cast(year_col)
-    month_col = _maybe_cast(month_col)
-    day_col = _maybe_cast(day_col)
-    hour_col = _maybe_cast(hour_col)
-    minute_col = _maybe_cast(minute_col)
-    second_col = _maybe_cast(second_col)
-    return parsing.try_parse_datetime_components(
-        year_col, month_col, day_col, hour_col, minute_col, second_col
+    """
+    Parse columns with datetime information into a single datetime column.
+
+    .. deprecated:: 1.1.0
+        Use pd.to_datetime({"year": year_col, "month": month_col, "day": day_col,
+        "hour": hour_col, "minute": minute_col, second": second_col}) instead to get
+        a Pandas Series.
+        Use ser = pd.to_datetime({"year": year_col, "month": month_col, "day": day_col,
+        "hour": hour_col, "minute": minute_col, second": second_col}) and
+        np.array([s.to_pydatetime() for s in ser]) instead to get a Numpy array.
+    """
+
+    warnings.warn(
+        """
+        Use pd.to_datetime({"year": year_col, "month": month_col, "day": day_col,
+        "hour": hour_col, "minute": minute_col, second": second_col}) instead to get a Pandas Series.
+        Use ser = pd.to_datetime({"year": year_col, "month": month_col, "day": day_col,
+        "hour": hour_col, "minute": minute_col, second": second_col}) and
+        np.array([s.to_pydatetime() for s in ser]) instead to get a Numpy array.
+""",  # noqa: E501
+        FutureWarning,
+        stacklevel=2,
+    )
+
+    ser = to_datetime(
+        {
+            "year": year_col,
+            "month": month_col,
+            "day": day_col,
+            "hour": hour_col,
+            "minute": minute_col,
+            "second": second_col,
+        }
     )
 
+    return np.array([s.to_pydatetime() for s in ser])
+
 
 def generic_parser(parse_func, *cols):
+    """
+    Use dateparser to parse columns with data information into a single datetime column.
+
+    .. deprecated:: 1.1.0
```

Comment:
these all should be 1.2 

its ok to just have the warning text, no need to double in the doc-string.

Reply:
Done